src/core/core.scale.js

```diff
@@ -274,26 +281,22 @@ class Scale extends Element {
 		this.min = undefined;
 		this.max = undefined;
 		/** @type {object[]} */
-		this.ticks = null;
+		this.ticks = [];
 		/** @type {object[]|null} */
 		this._gridLineItems = null;
 		/** @type {object[]|null} */
 		this._labelItems = null;
 		/** @type {object|null} */
 		this._labelSizes = null;
-		/** @type {number} */
-		this._length = undefined;
-		/** @type {object} */
+		this._length = -1;
 		this._longestTextCache = {};
-		/** @type {number} */
-		this._startPixel = undefined;
-		/** @type {number} */
-		this._endPixel = undefined;
-		this._reversePixels = undefined;
+		this._startPixel = -1;
+		this._endPixel = -1;
+		this._reversePixels = false;
 		this._userMax = undefined;
 		this._userMin = undefined;
-		this._ticksLength = undefined;
-		this._borderValue = undefined;
+		this._ticksLength = -1;
+		this._borderValue = -1;
```

Comment:
`0` is valid default for dimensions (pixel values and array sizes) IMO
I'd rather use typedefs and `undefined` with other undefined than `-1`

Reply:
done